p2p/server.py

```diff
@@ -355,8 +355,21 @@ def _make_peer_pool(self, discovery: DiscoveryProtocol) -> PeerPool:
             inbound=True,
         )
 
+        total_peers_inbound = 0.0
+        total_peers = 0.0
+
+        for current_peers in self.peer_pool.connected_nodes.values():
```

Comment:
Same here for avoiding mutability.  Here's some untested code that will do this a bit cleaner with less mutation.  You'll notice the approach is very similar.

```python
import operator
from cytoolz import groupby

total_peer_count = len(self.peer_pool)
inbound_peer_count = len([peer for peer in self.peer_pool.peers if peer.inbound])

if total_peer_count > 1 and inbound_peer_count // total_peers > 0.75:
    ...
```